---
title: "Why factorial designs?"
author: "Jeffrey A. Walker"
date: "12/27/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

# wrangling packages
library(here)
library(janitor)
library(readxl)
library(data.table)
library(stringr)

# analysis packages
library(MASS) # glm.nb
library(nlme)
library(lmerTest)
library(emmeans)
library(boot)
library(glmmTMB)
library(car)
library(DHARMa)

# graphing packages
library(ggsci)
library(ggpubr)
library(ggforce)
library(cowplot)

here <- here::here
source_path <- here("R/ggplotsci.R")
source(source_path)

clean_names <- janitor::clean_names

data_folder <- "data"

```

# Example 1 -- (sox10-) The enteric nervous system promotes intestinal health by constraining microbiota composition
## Import
```{r import-sox10-}
data_from <- "The enteric nervous system promotes intestinal health by constraining microbiota composition"
file_name <- "journal.pbio.2000689.s008.xlsx"
file_path <- here(data_folder, data_from, file_name)

# Fig 6d
fig6d_wide <- read_excel(file_path,
                    sheet = "Figure 6",
                    range = "I2:L16") %>%
#  clean_names()  %>%
  data.table()

group_levels <- colnames(fig6d_wide)
fig6d <- melt(fig6d_wide,
              measure.vars = group_levels,
              variable.name = "group",
              value.name = "count") %>%
  na.omit() # delete blank rows
fig6d[, group := factor(group, group_levels)]

# create two factors
fig6d[, genotype := ifelse(group == "WT" | group == "WT transplant",
                           "WT",
                           "sox10-")]
fig6d[, genotype := factor(genotype, levels = c("WT", "sox10-"))]
fig6d[, treatment := ifelse(group == "WT" | group == "sox10-",
                           "Control",
                           "Transplant")]
fig6d[, treatment := factor(treatment, levels = c("Control", "Transplant"))]

# View(fig6d)
```

## model and interpretation
```{r sox10- model and plot}
m1 <- glm.nb(count ~ treatment*genotype, data = fig6d)
m1_emm <- emmeans(
  m1,
  specs = c("treatment", "genotype"),
  type = "response"
)
m1_simple <- contrast(
  m1_emm,
  method = "revpairwise",
  simple = "each",
  combine = TRUE,
  adjust = "none"
)
gg1 <- plot_the_response(m1, m1_emm, m1_simple)

# make sox10- the ref so add sox10
fig6d[, sox10 := ifelse(genotype == "WT", "sox10+", "sox10-")]
fig6d[, sox10 := factor(sox10,
                        levels = c("sox10-", "sox10+"))]
m1 <- glm.nb(count ~ treatment*sox10, data = fig6d)
m1_emm <- emmeans(
  m1,
  specs = c("treatment", "sox10"),
  type = "response"
)
m1_simple <- contrast(
  m1_emm,
  method = "revpairwise",
  simple = "each",
  combine = TRUE,
  adjust = "none"
)
gg2 <- plot_the_response(m1, m1_emm, m1_simple)

```

## Variables

1. WT/control -- Negative control. Expected to have a low neutrophil count due to intact ENS
2. sox10-/control -- Positive control. known to have a high neutrophil count, hypothesis is, this is due to missing ENS.
3. WT/transplant -- methodological control for the transplant procedure. If the procedure has a trivial effect then we expect low neutrophil count (the same as WT)
4. sox10-/transplant -- This is the treatment of interest. If ENS regulates inflammation-inducing bacteria, then we expect partial return to WT level of neutrophils. If sufficient, we expect complete return to WT level of neutrophils.

## Conclusion from researchers

"Following transplantation, the sox10 mutants that developed a normal- appearing ENS extending along the entire length of the intestine had WT levels of intestinal neutrophils (Fig 6C and 6D), demonstrating that the ENS is sufficient to prevent intestinal inflammation."

## Plot
```{r sox10- model and plot}
plot_grid(gg1, gg2, ncol = 2)
```



## contrasts of interest
(2 - 1) -- the pattern to explain by a hypothesis. The hypothesis is that sox10- -> inflammation. The mediators are sox10- -> no ENS -> no regulation of certain bacteria -> inflammation
(3 - 1) -- effect of the transplant. If positive, then transplant-ENS interferes with sox10-ENS development or introduces some other path to inflammation. If negative, then transplant-ENS and sox10-ENS add. If zero, then transplant-ENS and sox10-ENS are redundant. A good way to think about this is the right side plot above where sox10 is added (group 2), transplant is added (group 3) and both are added (group 4).
(4 - 2) -- treatment of interest vs. positive control. This is main comparison. It is the absolute recovery. The hypothesis is this has a big effect. The interpretation is confounded by (3 - 1) the methodological effect. If (3 - 1) is positive because transplant increases inflammation (even though ENS itself is mitigating) or because transplanted ENS not as effective as naturally acquired ENS, then we'd expect (4 - 2) to be smaller.
(4 - 1) -- comparing effect of treatment vs. negative control but the interpretation is confounded by the transplant
(4 - 3) -- comparing effect of treatment vs method control. This measures the relative recovery. (4 - 2)/(4 - 3) is percent recovery (4 - 3) - (4 - 2)
 
## scenarios
1. transplant and sox10+ are redundant and equal.
* if WT is ref: generated by setting beta_4 = -beta_2. mu_1 = mu_3 = mu_4
* if sox10- is ref then generated by beta_sox10+ = beta_transplant = -beta_ixn. . mu_2 = mu_3 = mu_4
2. transplant-derived ENS or sox10+ derived ENS are redundant and equal but transplant introduces other issues that reduce effect by 1/2.
mu = b1 + b2 x geno + b3 x trans_ens + b4 * ixn + b5 x _trans

The interaction correctly estimates the effect of the ENS in both 1 and 2. The treatment effect correctly estimates the effect of the ENS in 1 but not 2. Still unconfident about this statement though.

Ultimately I don't think measuring the interaction matters.


## What ifs

(actual): large, negative interaction with 4 ~= 3
1. small negative interaction with 4 > 3. Incomplete recovery.
 - variant 1: 4 v. 2 > 0.05. Could interpret this as "no recovery". Simulate this.

This is key this is where the interaction is necessary for correct inference while pairwise comparisons are more ambiguous. Find a good data example with this.

Also suggests a way of adding interaction to the plot. Plot the additive value at +/+ and create a vertical bracket with p-value
 
```{r sox10- simulation why interaction}
sox10_simulator <- function(
  n = 10,
  n_iter = 1000,
  beta = c(0, 1, -0.5, 0) # mean, genotype, transplant, ixn
){
  # n <- 10
  # n_iter <- 1000
  # beta <- c(0, 1, -0.5, 0)
  genotype_levels <- c("wt", "ko")
  treatment_levels <- c("sham", "transplant")
  fake_data <- data.table(
    genotype = factor(rep(genotype_levels, each = n*2),
                      levels = genotype_levels),
    treatment = factor(rep(rep(treatment_levels, each = n), 2),
                       levels = treatment_levels)
  )
  X <- model.matrix(~ genotype*treatment, data = fake_data)
  ixn_b <- numeric(n_iter)
  ixn_p <- numeric(n_iter)
  treatment_e <- numeric(n_iter)
  treatment_p <- numeric(n_iter)
  for(i in 1:n_iter){
    set.seed(i)
    fake_data[, y := (X %*% beta)[, 1] + rnorm(n*2*2, mean = 0, sd = 1)]
    fit <- lm(y ~ genotype*treatment, data = fake_data)
    ixn_b[i] <- coef(summary(fit))["genotypeko:treatmenttransplant", "Estimate"]
    ixn_p[i] <- coef(summary(fit))["genotypeko:treatmenttransplant", "Pr(>|t|)"]
    fit_pair <- emmeans(fit, specs = c("genotype", "treatment")) %>%
      contrast(method = "revpairwise", adjust = "none") %>%
      summary() %>%
      data.table()
    treatment_e[i] <- fit_pair[5, estimate]  
    treatment_p[i] <- fit_pair[5, p.value]  
  }
  return(data.table(ixn_b = ixn_b,
                    ixn_p = ixn_p,
                    treatment_e = treatment_e,
                    treatment_p = treatment_p))
}
```


```{r sox10- simulation scenarios}
n <- 10
n_iter <- 1000

# complete recovery
beta1 <- c(0, 1, 0, -1)
sim1 <- sox10_simulator(n, n_iter, beta = beta1)

# no recovery
beta2 <- c(0, 1, 0, 0)
sim2 <- sox10_simulator(n, n_iter, beta = beta2)

# no recovery, -transplant effect
beta3 <- c(0, 1, -0.5, 0)
sim3 <- sox10_simulator(n, n_iter, beta = beta3)

# complete recovery, +transplant effect
beta4 <- c(0, 1, 0.5, -1)
sim4 <- sox10_simulator(n, n_iter, beta = beta4)

```

```{r sox10 sim_table}

sim_stats <- function(sim){
  stat_1_cases <- which(sim$ixn_p < 0.05 & sim$treatment_p > 0.05)
  stat_2_cases <- which(sim$ixn_p < 0.05)
  stat_3_cases <- which(sim$treatment_p < 0.05)
  stat_4_cases <- which(sim$ixn_p < 0.05 | sim$treatment_p < 0.05)
  stat_1 <- length(stat_1_cases)/n_iter
  stat_2 <- length(stat_2_cases)/n_iter
  stat_3 <- length(stat_3_cases)/n_iter
  stat_4 <- length(stat_4_cases)/n_iter
  return(c(stat_1, stat_2, stat_3, stat_4))
}

stats_table <- rbind(sim_stats(sim1),
                    sim_stats(sim2),
                    sim_stats(sim3),
                    sim_stats(sim4))
colnames(stats_table) <- c("b4+ & treat-",
                         "b4+",
                         "treat+",
                         "b4+ | treat+")
b_table <- rbind(beta1, beta2, beta3, beta4)
colnames(b_table) <- paste0("beta_", 1:4)

sim_table <- data.table(
  cbind(b_table, stats_table)
)

sim_table

```
1. case 1: transplant and sox10+ are redundant and equal.
* if WT is ref: generated by setting beta_4 = -beta_2
* if sox10- is ref then generated by beta_sox10+ = beta_transplant = -beta_ixn
2. case 2: transplant has no effect
3. case 3: transplant and sox10+ are additive (think of sox10- as ref)
4. case 4: transplant adds 1/2 inflammation. transplant ENS and sox10+ ENS are redundant

We don't know the truth. If we interpret treatment effect (4-2) as evidence of hypothesis then
1. case 1 - optimal (more power than ixn)
2. case 2 - equal to ixn
3. case 3 - 


```{r sox10- plot simulation}
i <- 40
set.seed(i)
fake_data[, y := (X %*% beta)[, 1] + rnorm(n*2*2, mean = 0, sd = 1)]
fit <- lm(y ~ treatment*genotype, data = fake_data)
fit_coef <- coef(summary(fit))
fit_emm <- emmeans(fit, specs = c("treatment", "genotype"))
fit_simple <- contrast(fit_emm,
                       method = "revpairwise",
                       simple = "each",
                       combine = TRUE,
                       adjust = "none") %>%
    summary(infer = TRUE)
fit_coef
plot_the_response(fit, fit_emm, fit_simple)
```
 
seed = 40 interesting. Small, positive wt/transplant+ effect, small interaction but nothing is p < 0.05 but interaction.
similar: 44, 63

# Example 2 -- (zeb1) Endothelial ZEB1 promotes angiogenesis-dependent bone formation and reverses osteoporosis

Fig 6. Reported on p. 6 second column. "Zeb1ΔEC mice with r.Dll4 protein remarkably restored/enhanced/improved/normalized". Zeb1ΔEC are endothelial-cell (EC) specific Zeb1 knockouts.

Good example of a terrible figure. Flattened with 4 flat labels. No indication of factorial design.

```{r import zeb1, warning = FALSE, message=FALSE}
data_from <- "Endothelial ZEB1 promotes angiogenesis-dependent bone formation and reverses osteoporosis"
file_name <- "190640_2_data_set_4272441_q1zm04.xlsx"
file_path <- here(data_folder, data_from, file_name)

# Fig 6d
fig6d_wide <- read_excel(file_path,
                    sheet = "Fig 6",
                    range = "B2:E7",
                    col_types = c("numeric",
                                  "numeric",
                                  "numeric",
                                  "numeric")) %>%
  clean_names()  %>%
  data.table()
colnames(fig6d_wide) <- c("wt_vehicle",
                          "wt_dll4",
                          "ko_vehicle",
                          "ko-dll4")
fig6d_wide[, experiment := factor(paste("exp_",.I))]
fig6d <- melt(fig6d_wide,
              id.vars = "experiment",
              variable.name = "treatment",
              value.name = "type_h")
genotype_levels <- c("wt","ko")
fig6d[, genotype := substr(treatment, 1, 2) %>%
        factor(genotype_levels)]
dll4_levels <- c("vehicle","dll4")
fig6d[, dll4 := 
        substr(treatment, 4, nchar(as.character(treatment))) %>%
        factor(dll4_levels)]
```

1. wt/vehicle -- Negative control. Expected to have normal vessel density
2. wt/dll4 -- methodological control. Expected to have normal vessel density if no effect when zeb1 present. 
3. ko/vehicle -- positive control. Expected to have low vessel density because of previous experiments showing this
4. ko/dll4 -- the treatment of interest. If it fully reverses ko, then no effect with dll4.

why interaction -- contrast (4 - 3) - the main comparison, but this has to be compared to (2 - 1). If (4-3) = (2-1) then effect of dll4 is not dependent on zeb1. Hence we need the interaction effect. This is the effect of dll4 conditional on zeb1.

```{r zeb1-models}
fig6d_m1 <- lm(type_h ~ genotype*dll4, data = fig6d)
coef(summary(fig6d_m1))

fig6d_m2 <- lmer(type_h ~ genotype*dll4 + (1|experiment), data = fig6d)
coef(summary(fig6d_m2))

```

Again its not good enough to show ko/dll4 is different from ko/vehicle. The interaction indicates that dll4 physically interacts with zeb1 in some way. It has something to do with zeb1 regulation of angiogenesis.

This suggests another simulation -- when is ko/dll4 > ko/vehicle AND
wt/vehicle > ko/vehicle (the two comparisons in fig6d) BUT there is no interaction?

 
```{r zeb1- simulation why interaction}
n <- 10
genotype_levels <- c("wt", "ko")
treatment_levels <- c("vehicle", "dll4")
fake_data <- data.table(
  genotype = factor(rep(genotype_levels, each = n*2),
                    levels = genotype_levels),
  treatment = factor(rep(rep(treatment_levels, each = n), 2),
                     levels = treatment_levels)
)
X <- model.matrix(~ genotype*treatment, data = fake_data)
# colnames(X)
beta <- c(0, -1, 0, 0)
n_iter <- 1000
ixn_p <- numeric(n_iter)
treatment_p <- numeric(n_iter)
control_p <- numeric(n_iter)
treatment_effect <- numeric(n_iter)
for(i in 1:n_iter){
  set.seed(i)
  fake_data[, y := (X %*% beta)[, 1] + rnorm(n*2*2, mean = 0, sd = 1)]
  fit <- lm(y ~ genotype*treatment, data = fake_data)
  ixn_p[i] <- coef(summary(fit))[4, "Pr(>|t|)"]
  fit_emm <- emmeans(fit, specs = c("genotype", "treatment"))
  fit_pair <- fit_emm %>%
    contrast(method = "revpairwise", adjust = "none") %>%
    summary() %>%
    data.table()
  treatment_p[i] <- fit_pair[5, p.value]  
  control_p[i] <- fit_pair[1, p.value]  
  treatment_effect[i] <- fit_pair[5, estimate]
}
which(ixn_p > 0.1 &
        treatment_p < 0.05 &
        control_p < 0.05 &
        treatment_effect > 0)

```

```{r zeb1 plot simulation}
i <- 180
set.seed(i)
fake_data[, y := (X %*% beta)[, 1] + rnorm(n*2*2, mean = 0, sd = 1)]
fit <- lm(y ~ genotype*treatment, data = fake_data)
fit_coef <- coef(summary(fit))
fit_emm <- emmeans(fit, specs = c("genotype", "treatment"))
fit_simple <- contrast(fit_emm,
                       method = "revpairwise",
                       simple = "each",
                       combine = TRUE,
                       adjust = "none") %>%
    summary(infer = TRUE)
fit_coef
plot_the_response(fit, fit_emm, fit_simple)
```

seed = 180 is perfect. Recovery? No reliable evidence that dll4 has anything to do with zeb1

## fig 6f  Tb. BV/TV
```{r import zeb1-6f Tb. BV/TV, warning = FALSE, message=FALSE}
data_from <- "Endothelial ZEB1 promotes angiogenesis-dependent bone formation and reverses osteoporosis"
file_name <- "190640_2_data_set_4272441_q1zm04.xlsx"
file_path <- here(data_folder, data_from, file_name)

# Fig 6f Tb. BV/TV
fig6f_wide <- read_excel(file_path,
                    sheet = "Fig 6",
                    range = "B18:E23",
                    col_types = c("numeric",
                                  "numeric",
                                  "numeric",
                                  "numeric")) %>%
  clean_names()  %>%
  data.table()
colnames(fig6f_wide) <- c("wt_vehicle",
                          "wt_dll4",
                          "ko_vehicle",
                          "ko-dll4")
fig6f_wide[, experiment := factor(paste("exp_",.I))]
fig6f <- melt(fig6f_wide,
              id.vars = "experiment",
              variable.name = "treatment",
              value.name = "tb_BV_TV")
genotype_levels <- c("wt","ko")
fig6f[, genotype := substr(treatment, 1, 2) %>%
        factor(genotype_levels)]
dll4_levels <- c("vehicle","dll4")
fig6f[, dll4 := 
        substr(treatment, 4, nchar(as.character(treatment))) %>%
        factor(dll4_levels)]
```

```{r zeb1-6f-models}
fig6f_m1 <- lm(tb_BV_TV ~ genotype*dll4, data = fig6f)
coef(summary(fig6f_m1))

```

# Example 3 -- (gba1) Gut-seeded alpha-synuclein fibrils promote gut dysfunction and brain pathology specifically in aged mice

Fig 2i

```{r gba1-import, warning = FALSE, message=FALSE}
data_from <- "Gut-seeded α-synuclein fibrils promote gut dysfunction and brain pathology specifically in aged mice"
file_name <- "41593_2020_589_MOESM6_ESM.xls"
file_path <- here(data_folder, data_from, file_name)

# Fig 2i
fig2i_wide <- read_excel(file_path,
                    sheet = "2I",
                    range = "A2:G5",
                    col_names = FALSE
                    ) %>%
#  clean_names()  %>%
  data.table()

colnames(fig2i_wide)[1:2] <- c("genotype", "treatment")
fig2i <- melt(fig2i_wide,
              id.vars = c("genotype", "treatment"),
              variable.name = "experiment",
              value.name = "gcase")
unique(fig2i[, genotype])
unique(fig2i[, treatment])
fig2i[, genotype := factor(genotype,
                           levels = c("WT", "ASO"))]
fig2i[, treatment := factor(treatment,
                            levels = c("GFP only", "GBA1"))]
fig2i[, group := paste(genotype, treatment, sep = "_")]

unique(fig2i[, group])
group_levels <- c("WT_GFP only", "WT_GBA1", "ASO_GFP only", "ASO_GBA1" )
fig2i[, group := factor(group,
                        levels = group_levels)]
qplot(x = group, y = gcase, data = fig2i)

```


α-Syn fibrals are bad. Glucocerebrosidase (GCase) is enzyme that maintains low α-Syn. GBA1 is gene for GCase. Thy1-α-Syn overexpressing (ASO) line is a transgenic model of synucleinopathy, with constitutive overproduction of α-Syn and chronically elevated p-α-Syn levels.

These findings highlight GBA1, the gene encoding GCase, as a therapeutic target for peripheral synucleinopathy. Gene transfer of GBA1 also resulted in a reduction in duodenal p-α-Syn at 60 days post-viral injection (d.p.v.i.) and partially restored the GI phenotype observed in ASO mice (Fig. 2i–k;

Response: GCase expression (relative to that in WT)

1. wt/gfp -- Negative control. Expected to have normal GCase
2. wt/gba -- positive control. Expected to have elevated GCase because gba
3. aso/gfp -- methodological control. Expected to have low gcase because aso
4. aso/gba -- the treatment of interest. If it inhibits aso mechanism, then full reversal, so ixn because of decrease GCase in 3. If no ixn, then recovery due to some mechanism independent of aso mechanism. This assumes (3-1) is negative.

why interaction -- contrast (4 - 3) - the main comparison, but this has to be compared to (2 - 1). If (4-3) = (2-1) then effect of gba is not dependent on aso. Hence we need the interaction effect. This is the effect of gba conditional on aso.

```{r gba1-models}
m1 <- lm(gcase ~ genotype*treatment, data = fig2i)
amc_model_check(m1)
spreadLevelPlot(m1)
```

```{r}
m2 <- glm(gcase ~ genotype*treatment,
          family = Gamma,
          data = fig2i)
m2_model_check <- simulateResiduals(fittedModel = m1, n = 250)
plot(m2_model_check, asFactor = F)

m3 <- gls(gcase ~ genotype*treatment,
          weights = varIdent(form = ~1|group),
          data = fig2i)
qqPlot(residuals(m3))
# heavy tails but probably better than heterogenity
# spreadLevelPlot(residuals(m3), by = group)

```

```{r}
m3_emm <- emmeans(m3, specs = c("genotype", "treatment"))
m3_simple <- contrast(m3_emm,
                      method = "revpairwise",
                      simple = "each",
                      combine = TRUE,
                      adjust = "none")
m3_ixn <- contrast(m3_emm, interaction = c("revpairwise"), by = NULL)

coef(summary(m1))
m3_simple
m3_ixn
```
Gene transfer of GBA1 also resulted in a reduction in duodenal p-α-Syn at 60 days post-viral injection (d.p.v.i.) and partially restored the GI phenotype observed in ASO mice

# Example 4 -- {XX mice} XX sex chromosome complement promotes atherosclerosis in mice

Interaction - Yes. Because we want to know XX effect conditional on sex. If big, then other things (other than XX) are contributing.

Source: [AlSiraj, Y., Chen, X., Thatcher, S.E., Temel, R.E., Cai, L., Blalock, E., Katz, W., Ali, H.M., Petriello, M., Deng, P. and Morris, A.J., 2019. XX sex chromosome complement promotes atherosclerosis in mice. Nature communications, 10(1), pp.1-13.](https://www.nature.com/articles/s41467-019-10462-z)

```{r xx-mice-import, message=FALSE}

data_folder <- "XX sex chromosome complement promotes atherosclerosis in mice"
file_name <- "41467_2019_10462_MOESM6_ESM.xlsx"
file_path <- here(data_path, data_folder, file_name)

# fig 3e
fig3e_wide <- read_excel(file_path,
                  sheet = "Figure 3E",
                  range = "A3:K4",
                  col_names = FALSE) %>%
  data.table() %>%
  transpose(make.names = 1)

sex_levels <- colnames(fig3e_wide)
fig3e <- melt(fig3e_wide,
              measure.vars = sex_levels,
              variable.name = "sex",
              value.name = "lesian_area")

# convert sex variable to factor
fig3e[, sex := factor(sex,
                        levels = sex_levels)]

# create chromosome column and convert to factor
chromosome_levels <- c("X", "Y_sry-")
fig3e[, chromosome := rep(rep(chromosome_levels, each = 5), 2)]
fig3e[, chromosome := factor(chromosome,
                             levels = chromosome_levels)]

# researchers treatment levels
fig3e[, treatment := rep(c("FXX", "FXY", "MXX", "MXY"), each = 5)]

# two rows with missing response so delte these rows as there is 
# no useful information in them
fig3e <- na.omit(fig3e) # be careful with a global na.omit

# View(fig3e) # highlight and run to see data
```

### Understand the experimental design

Hypothesis 1 - 2nd X increases atherosclerosis (for Fig3e, this is measured as lesion area)

**Response variable** (Y) -- atherosclerotic lesian area in aortic sinus. The hypothesis is that features on the X chromosome contribute to atherosclerosis

**Factor 1** -- *sex* ("F", "M"). This is not the typical sex factor that is merely observed but is a manipulated factor. Sex is determined by the presence or absence of *SRY* on an autosome using the Four Core Genotype mouse model. *SRY* determines the gonad that develops (ovary or testis). "F" does not have the autosome with *SRY*. "M" has the autosome with *SRY*.

**Factor 2** -- *chromosome* ("XX", "XY"). The chromosome complement is not observed but manipulated. In "XX", neither chromosome has *SRY* as the natural condition. In "XY", *SRY* has been removed from the Y chromosome. Alternative values might be ("X", "Y_sry-")

**Design** -- $2 \times 2$, mean two crossed factors each with two levels. This results in 4 groups, each with a unique combination of the levels from each factor. Two ways of thinking about this design are

1. "FXX" is the control. "MXX" has the added *SRY* gene. "FXY" replaces "X" with the Y_sry- complement. "MXY" has both the added *SRY* and the Y_sry- complement. In this system, we think of a positive response as a reduction in lesian area.
2. "FXY" is the control (it has neither *SRY* nor the complimentary X chromosome). "FXX" adds the X chromosome. "MXY" adds *SRY*. "MXX" adds both.

```{r xx-mice-factor-table, echo=FALSE}
factor_table <- data.table(
  treatment = c("FXX", "XX-male", "XY-female", "XY-male"),
  "2nd chromosome" = c("X", "X", "Y_sry-", "Y_sry-"),
  autosome = c("WT", "SRY+", "WT", "SRY+")
)

knitr::kable(factor_table) %>%
  kable_styling()
```

The role of the treatments helps us determine which contrasts to focus on

1. FXX -- the control or maybe better called the reference. Expect lesian area to be high based on prior knowledge. The hypothesis is, the high lesions are due to the 2nd X.
2. MXX -- if hypothesis true, then should see high lesion area. Diofference with FXX due to gonad sex.
3. FXY -- if hypothesis is true, then expect low lesion area compared to FXX
4. MXY -- if hypothesis is true, then expect low lesion area compared to MXX. Any difference with FXY due to gonad sex. 

This suggest the following contrasts

1. (FXY - FXX) -- effect of XX in mice with female gonad. If hypothesis is true, this should be large, negative.
2. (MXY - MXX) -- effect of XX in mice with male gonad. If hypothesis is true, this should be large, negative.
3. (MXY - MXX) - (FXY - FXX) -- Interaction. is the effect of XX conditional on gonad sex?

What about (MXX - FXX) -- the effect of gonadal sex given the XX background? Not a focal test.

What about (MXY - FXY) -- the effect of gonadal sex given the XY background? Certainly not a focal test.

Either of these can be thought of as used indirectly in computing the interaction although, technically, neither is actually used to compute the interaction.

### Fit the model {#xx-mice-fit}

```{r xx-mice-lm}
# step 1 - fit the model
m1 <- lm(lesian_area ~ sex*chromosome, data = fig3e)
```

```{r xx-mice-coef}
# step 2 - get the coefficient table
m1_coef <- cbind(coef(summary(m1)),
                 confint(m1))
m1_coef %>%
  kable(digits = c(1,1,2,4,1,1)) %>%
  kable_styling()
```

```{r xx-mice-emm}
# step 3 - get the modeled means
m1_emm <- emmeans(m1, specs = c("sex", "chromosome"))

m1_emm %>%
  summary() %>%
  kable(digits = c(0,1,2,0,2,2)) %>%
  kable_styling()
```

```{r xx-mice-planned}
# m1_emm # print in console to get row numbers
# set the mean as the row number from the emmeans table
fxx <- c(1,0,0,0)
mxx <- c(0,1,0,0)
fxy <- c(0,0,1,0)
mxy <- c(0,0,0,1)

# contrasts are the difference in the vectors created above
# the focal contrasts are in the understand the experiment section
# 1. (FXY - FXX) 
# 2. (MXY - MXX)
# 3. Interaction

m1_planned <- contrast(m1_emm,
                       method = list(
                         "FXY - FXX" = c(fxy - fxx),
                         "MXY - MXX" = c(mxy - mxx)
                       ),
                       adjust = "none"
) %>%
  summary(infer = TRUE)

m1_interaction <- contrast(m1_emm,
                           interaction = c("revpairwise"),
                           by = NULL) %>%
  summary(infer = TRUE)

# Make the interaction table conform (same column names)
# to the planned table
m1_interaction <- cbind(contrast = "interaction",
                        m1_interaction[, -c(1:2)]) # remove cols 1:2
# combine
m1_planned <- rbind(m1_planned, m1_interaction)

m1_planned %>%
  kable(digits = c(0,1,1,0,1,1,2,5)) %>%
  kable_styling()

```


```{r plot_the_model}

m1_plot <- ggplot_the_model(
  fit = m1,
  fit_emm = m1_emm,
  fit_pairs = m1_planned[1:3,],
  palette = pal_okabe_ito_blue,
  legend_position = "bottom",
  y_label = expression(paste("Lesian area (", mu, m^2, ")")),
  effect_label = expression(paste("Effect (", mu, m^2, ")")),
  contrast_rows = "all"
)
m1_plot
```

